pkg/webhook/clusterset/validatingWebhook.go

```diff
@@ -163,6 +164,11 @@ func (a *AdmissionHandler) validateUpdateRequest(request *v1.AdmissionRequest) *
 			return rejectStatus
 		}
 
+		//For upgrade from 2.4 to 2.5, we need to move all clusters from empty set to "default" set.
+		if currentClusterSetName == defaultSetName {
+			return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
```

Comment:
I think this should only be bypassed when the cluster is from the claim.  It should be like in line 188

```go
// the cluster is claimed from clusterpool, do not allow to update clustersetlabel.
if clusterDeployment == nil || clusterDeployment.Spec.ClusterPoolRef == nil {
	return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
}

if len(originalClusterSetName) == 0 && currentClusterSetName == defaultSetName {
    return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
}

rejectStatus.Result.Message = "Do not allow to update claimed cluster's clusterset."
return rejectStatus

```

Reply:
done